channeldb/payments.go

```diff
@@ -460,6 +460,12 @@ type PaymentsQuery struct {
 	// true, the query will fetch payments with indices lower than the
 	// IndexOffset.
 	Reversed bool
+
+	// If IncludeIncomplete is true, then return payments that have
+	// not yet fully completed. This means that pending payments,
+	// as well as failed payments will show up if this
+	// field is set to True.
+	IncludeIncomplete bool
```

Comment:
I think that this commit belongs before we introduce paginated queries? That way you don't need to touch this code twice, and the commit where you add pagination works as intended from the get go.

Reply:
I kept this commit separated, as I was not sure whether the incomplete payments should be discarded at the level of QueryPayments or at the rpcserver level and wanted to discuss this. I think it's best as it is now and also squashed the commit with the payments query commit as you suggest.